src/python/pants/notes/2.20.x.md

```diff
@@ -0,0 +1,180 @@
+
+The new [`local_scheme`](https://www.pantsbuild.org/2.20/reference/targets/vcs_version#local_scheme) and [`version_scheme`](https://www.pantsbuild.org/2.20/reference/targets/vcs_version#version_scheme) fields on the `vcs_version` target allow additional customization of its behaviour.
+
+Improvements to dependency inference:
+
+- [Strings imports](https://www.pantsbuild.org/2.20/reference/subsystems/python-infer#string_imports) are handled more reliably, including obeying `pants: no-infer-dep` comments and having fewer false positives on invalid module names.
+- Django-specific dependency inference (via the `pants.backend.experimental.python.framework.django` backend) now infers migrations and management commands as dependencies of `apps.py`.
+
+[The `pants.backend.codegen.protobuf.python` backend](https://www.pantsbuild.org/2.20/docs/python/integrations/protobuf-and-grpc) now supports [using `grpclib`](https://www.pantsbuild.org/2.20/docs/python/integrations/protobuf-and-grpc#use-alternative-grpc-plugins) in addition to `grpcio` for generating GRPC service stubs.
+
+Default module mappings were added for more modules: `django-countries`, `django-fsm`, `django-object-actions`, `django-postgres-extra`, `django-redis`, `django-scim2`, `djangorestframework-api-key`, `djangorestframework-queryfields`, `google-api-python-client`, `google-auth`, `grpcio-health-checking`, `grpcio-reflection`, `honeycomb-opentelemetry`, `opencv-python-headless`, `opentelemetry-sdk` (expanded to cover more modules). The special `_typeshed` module no longer has errors if imported.
+
+`pip list` now behaves correctly in venvs created via `export` with [the `[export].py_editable_in_resolve` option](https://www.pantsbuild.org/2.20/reference/goals/export#py_editable_in_resolve) enabled.
+
+There is now documentation on how to [exclude files from Black or isort](https://www.pantsbuild.org/2.20/docs/python/overview/linters-and-formatters#black-and-isort-excluding-files).
+
+The deprecation for the `[export].symlink_python_virtualenv` option has expired and it has been removed. Instead set [the `[export].py_resolve_format` option](https://www.pantsbuild.org/2.20/reference/goals/export#py_resolve_format) to `symlinked_immutable_virtualenv`.
+
+#### Terraform
+
+The Terraform backend now has [built-in support for lockfiles](https://www.pantsbuild.org/2.20/docs/terraform#lockfiles), using the `pants generate-lockfiles --resolve=path/to:module` goal, passing the address of the `terraform_module` target as the resolve.
+
+The default built-in version of Terraform has been upgraded from 1.4.6 to 1.7.1. Pants now has [built-in knowledge of versions up to 1.7.1](https://www.pantsbuild.org/2.20/reference/subsystems/download-terraform#known_versions), so the version can be overridden by setting [`[download-terraform].version`](https://www.pantsbuild.org/2.20/reference/subsystems/download-terraform#version) explicitly.
+
```

Comment:
we also did a fair rewrite of targets types, we should mention that here. Maybe:
```suggestion

The Terraform backend now has separate target types for vars files and backend configs, and handles these as normal dependencies in the `dependencies` field. See the new [instructions for Terraform deployments](https://www.pantsbuild.org/2.20/docs/terraform#deployments).

```

Reply:
Thanks! I added some more links and reworded this slightly.